lake/seekindex/writer.go

```diff
@@ -22,22 +24,31 @@ func NewWriter(w zio.Writer) *Writer {
 	}
 }
 
-func (w *Writer) Write(key zed.Value, offset int64) error {
+func (w *Writer) Write(key zed.Value, number uint64, offset int64) error {
 	b := w.builder
 	b.Reset()
 	if zed.IsContainerType(key.Type) {
 		b.AppendContainer(key.Bytes)
 	} else {
 		b.AppendPrimitive(key.Bytes)
 	}
+	b.AppendPrimitive(zed.EncodeUint(number))
 	b.AppendPrimitive(zed.EncodeInt(offset))
 	if w.typ != key.Type {
 		var schema = []zed.Column{
 			{"key", key.Type},
+			{"record", zed.TypeUint64},
```

Comment:
Since we're no longer restricted to indexing records, maybe call this count or index (in the sense of an array index)?